src/pipelinedefinition.cpp

```diff
@@ -85,14 +87,66 @@ Status PipelineDefinition::validate(ModelManager& manager) {
     return validationResult;
 }
 
+Status PipelineDefinition::initializeNodeResources() {
+    for (const auto& nodeInfo : nodeInfos) {
+        if (nodeInfo.kind == NodeKind::CUSTOM) {
+            void* customNodeLibraryInternalManager = nullptr;
+            auto it = nodeResources.find(nodeInfo.nodeName);
+            bool internalManagerExists = (it != nodeResources.end());
+            if (internalManagerExists) {
+                customNodeLibraryInternalManager = nodeResources.at(nodeInfo.nodeName);
+            }
+            auto params = createCustomNodeParamArray(nodeInfo.parameters).get();
+            if (!nodeInfo.library.isValid()) {
+                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Pipeline: {} node: {} refers to incorrect library", pipelineName, nodeInfo.nodeName);
+                return StatusCode::PIPELINE_DEFINITION_INVALID_NODE_LIBRARY;
+            }
+            auto status = nodeInfo.library.initialize(&customNodeLibraryInternalManager, params, nodeInfo.parameters.size());
```

Comment:
For pipelines with multiple, the same custom nodes we call initialize multiple times. Is it desired?

Reply:
Node library implementation should check if it needs to deinitialize & initialize again in case of reload.

Reply:
@dkalinowski it is expected behaviour -> each node manages its own parameters which may affect eg. buffer size, end-points so it is by design to have resource per node.